pkg/operator/ceph/cluster/mon/service.go

```diff
@@ -74,16 +74,16 @@ func (c *Cluster) createService(mon *monConfig) (*v1.Service, error) {
 }
 
 func (c *Cluster) exportService(service *v1.Service, monDaemon string) (string, error) {
+	// remove mon canary deployment only after the service is exported because DNS
```

Comment:
I see that it's a good thing to defer the deletion of the canaries to ensure they get cleaned up, but I don't see how this comment is related. The ordering of deletion of the canaries is not really changing, since anyway they were getting deleted at the end of this method (as long as no error on line 84).

Reply:
There was indeed an error on line 84 due to which deletion of the canary pod was deferred. 

```
2025-03-25T06:27:01.567902992Z 2025-03-25 06:27:01.567801 E | op-mon: failed to failover mon "d". failed to export service "rook-ceph-mon-e": failed to export service "rook-ceph-mon-e": failed to create service export successfully for the service "rook-ceph-mon-e": no status conditions available in the exported service "rook-ceph-mon-e"

```

Opened an upstream issue for the PR with more details.

Reply:
I see now, how about just clarifying the comment something like this?
```suggestion
	// defer removing the mon canary deployment to after the service is exported because DNS
```

Reply:
done.